### ISSUE TITLE
Using Hubflow

### ISSUE THREAD

**Issue by therevoltingx** [ethsociety/learn-plasma#45]
# Using HubFlow

## Why?
Hi, our development team uses HubFlow and think it might be a good tool to use for this project. 

You don't *need* HubFlow to follow the proposed git flow, but it makes things much easier.

HubFlow also enforces things like tagging, changelogs, and releases. This makes it easier to develop things for the project, submit pull requests, and generate releases. We could also integrate things like commit squashing so that pull requests are compacted to a single commit [optional].

### Starting a New Feature
Basically, the idea is that you have a `develop` branch that's the main branch by default. Then if you're working on a new feature you create a branch executing: `git hf feature start my-feature`

### Submitting a Pull Request
Once the branch is ready for review a pull request (can be labeled WiP) is submitted against the default branch on github. Once the PR is reviewed and approved it is merged into `develop`.

A pull request must also update the ChangeLog describing the changes.

### Generating a Release
Once the release is ready the developer can tag it and merge it with master by doing: `git hf release start 0.1.0` depending on the version.

## Notes
This is just a suggestion based on how our team works with HubFlow. If it's too complex for this open source project please don't hesitate to close this issue.

## Resources
ChangeLog: https://keepachangelog.com/en/1.0.0/
HubFlow: https://datasift.github.io/gitflow/

**Comment by barrasso** [comment]
Thanks for this @therevoltingx. As this project matures, I think that having a bit of structure in place is a good idea. We could also ease into it by simply following the underlying git flow for now (e.g using `develop` as the default branch).

What do you think @kfichter ?

**Comment by therevoltingx** [comment]
@barrasso 
Ah that's a great idea! Yeah i think just using `develop` as the default branch would be a great way to go IMHO.

**Comment by kfichter** [comment]
I'm going to close this for now. I don't think we're scaling far enough to make this necessary yet.
